pkgs/native_assets_cli/lib/src/api/c_compiler_config.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'build_config.dart';
+
+/// The configuration for a C toolchain.
+abstract final class CCompilerConfig {
+  /// Path to a C compiler.
+  Uri? get cc;
```

Comment:
I know these abbreviations are quite common, but how about still using the full names as in the docs? `compiler`, `linker`, `archiver`...